lib/tasks/import.rake

```diff
@@ -65,5 +65,45 @@ namespace :import do
 
     puts 'Unternehmen erfolgreich importiert!'
   end
+
+  namespace :people do
+    desc 'Import tags from people'
+    task tags: [:environment] do
+      non_members_csv = Wagons.find('svse').root.join('db/seeds/production/non_members.csv')
+      raise unless non_members_csv.exist?
+
+      # Clear taggings table for idempotent updates
+      ActsAsTaggableOn::Tagging.delete_all
+
+      # Get tag id for non_member tagging
+      tag = ActsAsTaggableOn::Tag.find_or_create_by(name: 'Nicht Mitglied im Dachverband')
+
+      CSV.parse(non_members_csv.read,
+                headers: true,
+                header_converters: :symbol).each do |non_member_row|
+        non_member = Person.find_by(first_name: non_member_row[:first_name],
+                                    last_name: non_member_row[:last_name],
+                                    address: non_member_row[:address])
+
+        if non_member.nil?
+          puts "Person mit dem Namen: " +
+                 "#{non_member_row[:first_name]} #{non_member_row[:last_name]} "+
+                 "wurde nicht gefunden."
+          next
+        end
+
+        tag_hash = {
+          taggable_id: non_member.id,
+          tag_id: tag.id,
+          taggable_type: 'tag',
+          context: 'tags'
+        }
+
+        ActsAsTaggableOn::Tagging.upsert(tag_hash)
```

Comment:
Upsert funktioniert am besten, wenn man einen UNIQUE-index hat. Wir sollten hier prüfen, ob man das gleiche Tag wirklich mehrfach der gleichen Person geben kann oder soll. Ich vermute, dass hier (zumindest in unserem Fall) eigentlich ein Unique-Index fehlt.

Reply:
Nach kurzer Durchsicht stelle ich fest, dass ein Unique-Index gesetzt ist. Damit funktioniert das Upsert wie gewünscht.